### ISSUE TITLE
Low cost Apple-Airtag clones

### ISSUE THREAD

**Issue by Cyl0nius** [biemster/FindMy#14]
Apple-Airtag clone with ST17H66 on Aliexpress (https://www.aliexpress.com/item/1005004495296995.html)

Runs with Open-Haystack-App with no problems.

The minimal assembly of the pcb is interesting. No capacitors at the xtal and no antenna matching at all (see schematic).

Flashed with STC Auto-Programmer (CH340) without any problem.
1. connect P10 to TXD, P9 to RXD, GND to GND
2. start flash script
3. connect 3.3V to Battery +

All connection points are exposed on PCB.

![IMG_20221206_105611](https://user-images.githubusercontent.com/119928546/206107327-24c16034-e232-4341-a88e-819e5c5dc92c.jpg)

![IMG_20221202_153814](https://user-images.githubusercontent.com/119928546/206107262-493b4d2e-ce21-4de6-ba96-a50812817188.jpg)

![IMG_20221202_153732](https://user-images.githubusercontent.com/119928546/206107408-03b2cd88-3ab4-4aea-8b35-f0291143df0a.jpg)

![schaltplan](https://user-images.githubusercontent.com/119928546/206107864-155dcaf4-5198-45fb-9256-e25ea11fd574.png)

**Comment by biemster** [comment]
Nice find! Following the link I noticed that I put it on my wishlist as well already a while ago, but never ordered it :)
Great that you took out the solder iron and started slashing to a minimum, does that mean there was an LED and a switch originally?
And what is the fifth test pad connected to?

**Comment by Cyl0nius** [comment]
More cheap tags available ...

![IMG_20221222_094816](https://user-images.githubusercontent.com/119928546/209045041-f3c78542-f133-4763-9970-b55d2320fcdd.jpg)

![IMG_20221222_094037](https://user-images.githubusercontent.com/119928546/209045157-76c5a13a-b1fc-453a-bb6f-e9ddaf86a26e.jpg)

![IMG_20221222_094101](https://user-images.githubusercontent.com/119928546/209045183-15ed357a-5c7d-438d-8ad3-a94cf84976aa.jpg)

**Comment by biemster** [comment]
Nice :+1: Those I have as well, in fact they were the first vadimkozhin managed to flash :1st_place_medal:

**Comment by Cyl0nius** [comment]
For easy programming the tags, even in bulk ... programming jig for 1mm pogo-pins.

STL is not supported, so I attach it as ZIP.

![IMG_20221230_125625](https://user-images.githubusercontent.com/119928546/210039499-afcdb2f0-5e81-4b05-8165-dac196674c5a.jpg)

[st17h66_programming_jig.zip](https://github.com/biemster/FindMy/files/10323270/st17h66_programming_jig.zip)

**Comment by drott** [comment]
> For easy programming the tags (button type as seen in first picture), even in bulk ... programming jig for 1mm pogo-pins.

Nice jig, thanks for sharing that. What are B and the button connected to? Do you need to keep the button pressed on power on or what do you do with it?

**Comment by Cyl0nius** [comment]
B = black = GND
The button is simply a switch for power supply. After(!) starting the flash-script just press the button and hold it until flash is done.
I'm running the flash-script on Windows10.
Just run CMD as administrator (otherwise USB communication will not work) and execute the script with:
python3 flash_st17h66.py
After(!) starting the flash-script wait for a second, connect power (3VCD) till end of flash.

**Comment by biemster** [comment]
> So far I failed to flash ST17H66B and C tags that I have here.

@drott How do you power your tags while flashing? They require more than some usb uarts can provide, as you might have already read in issue #5. Where does the flash script fail?

**Comment by drott** [comment]
> @drott How do you power your tags while flashing? They require more than some usb uarts can provide, as you might have already read in issue #5. Where does the flash script fail?

I do connect a separate 3.3V power supply (based on a RD DPS5005 and a Makita battery) to the 3v3 pad or clamped the battery + metal. The flash script starts running, first I hear the buzzer with less volume, then with more when I connect the power, but I don't get much serial data back from the to-be-flashed tag: Occassionally I see some '\x00" serial responses printed, but not getting to the required `cmd>>` (or similar). I know ordered some SOP16 programming testing clips to see if I can make it work with those.

**Comment by biemster** [comment]
Sorry I have to ask to eliminate, did you try swapping RX and TX? (I had them wrong way round for quite a while)

**Comment by Cyl0nius** [comment]
I just want to mention it. 

The chip must be complete powerless (no battery connected) when the script starts.
Supply power only after a few seconds after the start of the flashing script.

**Comment by aproxtimedev** [comment]
Sorry, i forgot to check this repo on folder Lenze_ST17H66

**Comment by ggaljoen** [comment]
New pcb version, works perfect! ![ST17H66C.png](https://user-images.githubusercontent.com/7492159/220330468-20275e78-1192-48ea-b92b-5f25f194f1c5.png)

Pogopins do their job fine: ![PogoPins-USB.jpg](https://user-images.githubusercontent.com/7492159/220330760-7705d0a5-5dcc-4ea2-8be5-192bc50b4790.jpg)

Added a switch for the power to the board.

**Comment by olivluca** [comment]
@ggaljoen  nice, but from the photo I'm not sure I understand which pogo pin touches which point (red, yellow and orange should be on the same line, yet one of them is slightly displaced), please tell me if my interpretation in the following photo is correct. 
Also, how do you keep the jig aligned?

![image](https://user-images.githubusercontent.com/8250980/220365097-6f7add88-b256-49f9-8eb9-400daab05290.png)

**Comment by ggaljoen** [comment]
> Also, how do you keep the jig aligned?

Alignment is free hand practice, like this: 
![pogo_in_action.jpg](https://user-images.githubusercontent.com/7492159/220379027-b6e8c820-1f64-42a0-8c74-093aabed2363.jpg)

Contact with P9 and 3V3 Vcc as guideline; 
![PogoPins.jpg](https://user-images.githubusercontent.com/7492159/220379576-88fa4849-0238-479e-9b93-0f89ca22a0b3.jpg)

**Comment by biemster** [comment]
I finally received my tags like in the first post of this issue, they look really nice! By far the best of all the versions I have, tiny and tight pcb. Let's see if I manage this weekend to flash them, did anybody here try those airtag cases / holders / keychain things on these?

**Comment by Pasarel** [comment]
Thank you Cyl0nius for the instructions, and biemster and vadimkozhin for the files in this repo. I flashed three devices like the ones in the first post with no issues, and they seem to be working fine atm.

Regarding those keychains sold on Aliexpress, I would advise against it. I bought one with my tags and it's too tight -- the tags don't fit. And looking through the reviews on Aliexpress, I am not the only person who faced this problem.

**Comment by biemster** [comment]
> Regarding those keychains sold on Aliexpress, I would advise against it. I bought one with my tags and it's too tight -- the tags don't fit. And looking through the reviews on Aliexpress, I am not the only person who faced this problem.

Thanks for reporting this. Is it the tags being a different size than the original airtags, or are the keychains just not good for either?

**Comment by Pasarel** [comment]
I haven't compared the size to the original yet, but I will in the next few days and I'll write back.

**Comment by big-mak** [comment]
> I haven't compared the size to the original yet, but I will in the next few days and I'll write back.

Any update? Did a proof of concept on one like @Cyl0nius has, now looking to buy a bunch of airtag lookalikes.

**Comment by vadimkozhin** [comment]
Just to confirm, that [this](https://www.aliexpress.com/item/1005004495296995.html) item (which was mentioned by @Cyl0nius) works. It has an original airtag formfactor and should be compatible with airtag accesories. The board have marked pads for TX and RX pins, so soldering will be a lot easier.

**Comment by gammadog808** [comment]
Hi guys,
   I'm trying to flash a ST17H66(in windows), looks to be the same from the first post.  When I run the flash program, wait a few seconds and apply 3v3 to the board, I get this error in python:

Traceback (most recent call last):
  File "C:\dit\FindMy-monterey\Lenze_ST17H66\flash_st17h66.py", line 80, in <module>
    res = uart.read(10)
  File "C:\dit\venv\lib\site-packages\serial\serialwin32.py", line 295, in read
    raise SerialException("GetOverlappedResult failed ({!r})".format(ctypes.WinError()))
serial.serialutil.SerialException: GetOverlappedResult failed (PermissionError(13, 'Access is denied.', None, 5))

Not sure why I get this error - I'm logged in as thee administrator, ran pycharm(or cmd) as admin, but no dice.  Even moved the default COM port on the usb flasher, still no luck.

Modified the COM line in the .py:
#uart = serial.Serial('/dev/ttyUSB0', 9600, timeout=0.01, inter_byte_timeout=0.01)
uart = serial.Serial('COM6', 9600, timeout=0.01, inter_byte_timeout=0.01)

Any ideas?

**Comment by biemster** [comment]
@gammadog808 this seems to be an issue with serial on your machine indeed, and not with the script. Please try to get that running first with a simple test program (connect tx directly to rx, and see if it echos the characters)
Also, please open a separate issue if you continue having problems.

**Comment by SergeKZ** [comment]
tell me where you can find the source Lenze_ST17H66

**Comment by danhuanggt** [comment]
> > @drott How do you power your tags while flashing? They require more than some usb uarts can provide, as you might have already read in issue #5. Where does the flash script fail?
> 
> I do connect a separate 3.3V power supply (based on a RD DPS5005 and a Makita battery) to the 3v3 pad or clamped the battery + metal. The flash script starts running, first I hear the buzzer with less volume, then with more when I connect the power, but I don't get much serial data back from the to-be-flashed tag: Occassionally I see some '\x00" serial responses printed, but not getting to the required `cmd>>` (or similar). I know ordered some SOP16 programming testing clips to see if I can make it work with those.

@drott Did you ever get your tags flashed? I'm encountering the same issue of not reaching the `cmd>>` and getting `\x00` responses back. Your experience mimics mine!

Here is how my device is hooked up to a CP2102.

![IMG_0760](https://github.com/biemster/FindMy/assets/920252/5e9f2dc4-c99c-407b-8dcf-cc2fa0b48a18)

**Comment by humpataa** [comment]
Maybe I should have opened another "issue" but it somehow belongs here, so ...
Do these chinese AirTag clones actually work like Apple's AirTags?
I have tried to change the manufacturer bytes of Apple (0x004c) to 0x0501 (cheap chinese iTag). But this results in the tag not being reported to Apple's server anymore. At least requested reports no longer include info for it anymore. I have not changed anything else, just the manufacturer bytes.
I can see using nRFConnect that the original iTag is using a much shorter advertising string – 15 bytes instead 30 bytes ...
Has anyone looked into this? How are they "using" Apple's network?
Or do I have to change the request for reports to get info about tags with different manufacturers?
Any hint is appreciated.

**Comment by biemster** [comment]
@humpataa This should go to another issue indeed, where you might explain your question in a bit more detail. Changing the manufacturer bytes would obviously cause the tag not to be reported anymore. Cheap chinese iTags do not participate in Apple's FindMy network (but it is possible I don't understand your question)

**Comment by humpataa** [comment]
Thanks for the quick reply. So what network do they use? Android, Samsung, Google – I didn't know that there actually IS someone else doing this like Apple does ...

**Comment by biemster** [comment]
those cheap iTags don't use any distributed network, the registered phone just remembers where it saw them last.

**Comment by humpataa** [comment]
oh really? that really sounds like "chinese" ... and makes them rather useless if not flashed. thank you!

**Comment by biemster** [comment]
Apple's FindMy network seems to be the only current solution, besides smaller networks that require dedicated apps like Tile or Chipolo. Although Android might join the game soon: https://www.zdnet.com/article/is-googles-find-my-device-network-for-android-nearing-settings-signs-point-to-yes/

**Comment by steve-m** [comment]
Unfortunately the tags from the first post are not available anymore. I ordered similar looking ones, but unfortunately they have the ST17H66T variant with OTP ROM, so they are useless. Does anyone have a source for tags with the B2 flash variant?

**Comment by isibizi** [comment]
> Unfortunately the tags from the first post are not available anymore. I ordered similar looking ones, but unfortunately they have the ST17H66T variant with OTP ROM, so they are useless. Does anyone have a source for tags with the B2 flash variant?

Are you sure that the T variant not working with script? @Cyl0nius do you have any news about this issue?

**Comment by zjonesz** [comment]
![image](https://github.com/user-attachments/assets/973371e4-1da7-4241-bad9-e69dda0c8fe6)

**Comment by liufuhu** [comment]
sh17h66T chip can not flash，66t is OTP chip，66b can relfash

**Comment by Systm21** [comment]
I asked Aaron if he had any ideas for a cheap but economical OHS transmitter. Aaron hacked Bluetooth and other devices and was actually able to help. He has written his own webflasher for the 1.50€ TB-03F. 

Have a look for yourself and support him, there is a lot to discover with him, such as the game Doom on a toothbrush. 

https://x.com/atc1441/status/1856136207501037916

**Comment by josemariaaraujo** [comment]
Hi, I've ordered a bunch (5 different styles/vendors) of low cost(<2€) bluetooth trackers from aliexpress hoping I could find one with ST17H66B, but all but one came with an umarked SOT8 that I suppose is the WS8300 (OTP) and the other one came with ST17H66T, also OTP.

Does anyone have a good source for a low cost tracker already with plastic case with battery compartment that can be reprogrammable to join the FindMy network?

Thank you.

**Comment by omarkhali** [comment]
> I asked Aaron if he had any ideas for a cheap but economical OHS transmitter. Aaron hacked Bluetooth and other devices and was actually able to help. He has written his own webflasher for the 1.50€ TB-03F.
> 
> Have a look for yourself and support him, there is a lot to discover with him, such as the game Doom on a toothbrush or OpenEpaperLink.
> 
> https://x.com/atc1441/status/1856136207501037916

Hello I try to flash firmware to Mi LYWSD03MMC,TB-03F  but there are no advertise of public key

**Comment by biemster** [comment]
> Hello I try to flash firmware to Mi LYWSD03MMC,TB-03F but there are no advertise of public key

Any other signs of life from the MCU? Did you flash this chip before?
I'll need some more info to be able to help, also please open a new issue for this.

**Comment by lovelyelfpop** [comment]
I was able to flash the hex file, but the tag not working, bluetooth seactch cannot found it, FindMy app cannot discover it. There is no response when I press the button on the board.

![11111111](https://github.com/user-attachments/assets/d91002d5-db34-4def-bfb8-38a9f541688c)

This is my board with a chip ST17h66B
![222222222](https://github.com/user-attachments/assets/b6db4389-5683-4576-a9e6-ed2b161ce3e0)

**Comment by biemster** [comment]
@lovelyelfpop  I'm going to ask you the same as omarkhali in literally the first comment above yours, please open a new ticket instead of overtaking someone else's.

**Comment by lovelyelfpop** [comment]
Sorry, it did work, I can see locations on android app.  The button on the board is useless after flash the firmware
